apps/api_web/lib/api_web/controllers/stop_controller.ex

```diff
@@ -240,7 +243,11 @@ defmodule ApiWeb.StopController do
   def show_data(conn, %{"id" => id} = params) do
     case Params.validate_includes(params, @show_includes, conn) do
       {:ok, _includes} ->
-        Stop.by_id(id)
+        case LegacyStops.expand([id], conn.assigns.api_version) do
+          [single] -> Stop.by_id(single)
+          [_, _] = pair -> Enum.find_value(pair, &Stop.by_id/1)
+          _ -> nil
+        end
```

Comment:
I'm beginning to think this isn't a great way to handle this. It only works in the specific scenario where a stop ID is renamed exactly once, and never split into multiple stops — if any splitting or additional renames happen later, it spontaneously starts returning 404 for a legacy ID that used to work fine. This is a fairly opaque behavior compared to just always returning 404 with no backwards compatibility, and feels likely to bite us later.

What we _could_ do is, in `LegacyStops`, encode the knowledge of whether a "new" stop ID replaces the old one, or should be used in addition to it (i.e. platforms). Then `expand` could have an "only renames" option which, if enabled, ignores IDs that are additions. In a situation like:

```
alpha => beta    => gamma
      \> beta-01 => gamma-01
      \> beta-02 => gamma-02
```

Expanding `[alpha]` normally would result in _all_ of the IDs above, but expanding it with "only renames" would just return `[alpha, beta, gamma]`, which would give correct behavior for this endpoint.

Reply:
I agree that this probably isn't the behavior we want, for the reasons you mention. 

The rename vs replace is an interesting idea. And actually, it probably matters for the `place-dudly` to `place-nubn` change, since that's a parent station change, and it feels bad to break that stop for people. But I'm ambivalent about the necessity of handling in the general case vs just a one-off case for that parent station.

Reply:
> it probably matters for the `place-dudly` to `place-nubn` change, since that's a parent station change, and it feels bad to break that stop for people.

That specific change does work fine with this code, since there are no "splits" and we've only renamed the stop once (so far?). Unfortunately it doesn't work with some other stop renames we already have, for example `South Station` to `NEC-2287`, since a "platform split" also occurred so the expanded set of stop IDs is more than `[old, new]`.